open_mastr/xml_download/utils_write_to_database.py

```diff
@@ -26,75 +27,81 @@ def write_mastr_xml_to_database(
+
+            df = read_xml_file(f, file_name)
+            df = process_table_before_insertion(
+                df,
+                xml_table_name,
+                zipped_xml_file_path,
+                bulk_download_date,
+                bulk_cleansing,
+            )
 
-                add_table_to_database(
-                    df=df,
-                    xml_tablename=xml_tablename,
-                    sql_tablename=sql_tablename,
-                    if_exists="append",
-                    engine=engine,
+            if engine.dialect.name == "sqlite":
+                add_table_to_sqlite_database(df, xml_table_name, sql_table_name, engine)
+            else:
+                add_table_to_non_sqlite_database(
+                    df, xml_table_name, sql_table_name, engine
                 )
+
     print("Bulk download and data cleansing were successful.")
 
 
-def is_table_relevant(xml_tablename: str, include_tables: list) -> bool:
+def extract_xml_table_name(file_name: str) -> str:
+    """Extract the XML table name from the file name."""
+    return file_name.split("_")[0].split(".")[0].lower()
+
+
+def extract_sql_table_name(xml_table_name: str) -> str:
+    """Extract the SQL table name from the xml table name."""
+    return tablename_mapping[xml_table_name]["__name__"]
+
+
+def is_table_relevant(xml_table_name: str, include_tables: list) -> bool:
     """Checks if the table contains relevant data and if the user wants to
     have it in the database."""
     # few tables are only needed for data cleansing of the xml files and contain no
     # information of relevance
     try:
         boolean_write_table_to_sql_database = (
-            tablename_mapping[xml_tablename]["__class__"] is not None
+            tablename_mapping[xml_table_name]["__class__"] is not None
         )
     except KeyError:
         print(
-            f"Table '{xml_tablename}' is not supported by your open-mastr version and "
+            f"Table '{xml_table_name}' is not supported by your open-mastr version and "
             f"will be skipped."
         )
         return False
     # check if the table should be written to sql database (depends on user input)
-    include_count = include_tables.count(xml_tablename)
+    include_count = include_tables.count(xml_table_name)
 
     return include_count == 1 and boolean_write_table_to_sql_database
 
 
-def create_database_table(engine: sqlalchemy.engine.Engine, xml_tablename: str) -> None:
-    orm_class = tablename_mapping[xml_tablename]["__class__"]
-    # drop the content from table
-    orm_class.__table__.drop(engine, checkfirst=True)
-    # create table schema
-    orm_class.__table__.create(engine)
+def create_database_table(
+    engine: sqlalchemy.engine.Engine, xml_table_name: str
+) -> None:
+    """Create the table in the database if it does not exist."""
+    orm_class = tablename_mapping[xml_table_name]["__class__"]
+    if not inspect(engine).has_table(orm_class.__tablename__):
```

Comment:
I think I would keep the old process of:
1. deleting the old table
2. creating a new empty table
3. Filling this empty table with new data

Reason:
They can change units, eg change the value of `power` or so. If we do not overwrite old data, we will not have this change in our database. We could of course also have a combined Primary Key together with the data column indicating recent changes to the data, but I think this adds more complexity then necessary. I would argue for keeping this code:

```python
orm_class = tablename_mapping[xml_tablename]["__class__"]
orm_class.__table__.drop(engine, checkfirst=True)
orm_class.__table__.create(engine)
```

Reply:
Thanks a lot for explaining! I wasn't aware that this situation could happen. It makes a lot of sense then to recreate the table. I updated the implementation as you suggested